src/scripts/main.js

```diff
@@ -1,3 +1,368 @@
+const getTBody = document.querySelectorAll('tbody tr');
+const getTFooter = document.querySelectorAll('tfoot tr');
+let clickCount = 0;
+let lastClickedColumn = null;
+
+// ASC sort
+function sortASC(columnName, cell) {
+  const newArr = [];
+
+  getTBody.forEach((item) => {
+    const itemContent = item.cells[cell];
+
+    newArr.push(itemContent.textContent);
+  });
+
+  if (columnName === 'Salary') {
+    newArr.sort(
+      (a, b) =>
+        transformIntoNumberFromSalary(a) - transformIntoNumberFromSalary(b),
+    );
+  } else {
+    newArr.sort();
+  }
+
+  // Оновлюємо таблицю
+  getTBody.forEach((item, index) => {
+    const itemContent = item.cells[cell];
+
+    itemContent.textContent = newArr[index];
+  });
+}
+
+// Функція для перетворення зарплати в число
+function transformIntoNumberFromSalary(str) {
+  return +str.replace(/[^0-9.-]+/g, '');
+}
+
+// DESC sort
+function sortDESC(columnName, cell) {
+  const newArr = [];
+
+  getTBody.forEach((item) => {
+    const itemContent = item.cells[cell];
+
+    newArr.push(itemContent.textContent);
+  });
+
+  if (columnName === 'Salary') {
+    newArr
+      .sort(
+        (a, b) =>
+          transformIntoNumberFromSalary(a) - transformIntoNumberFromSalary(b),
+      )
+      .reverse();
+  } else {
+    newArr.sort().reverse(); // Сортуємо і реверсуємо
+  }
+
+  // Оновлюємо таблицю
+  getTBody.forEach((item, index) => {
+    const itemContent = item.cells[cell];
+
+    itemContent.textContent = newArr[index];
+  });
+}
+
+
+getTableHeader.forEach((header) => {
+  // eslint-disable-next-line no-shadow
+  header.addEventListener('click', function (event) {
+    const getClickOnHeader = event.target.closest('th');
+
+    if (getClickOnHeader) {
+      const columnName = header.textContent;
+
+      // Скидаємо clickCount, якщо натискаємо на нову колонку
+      if (lastClickedColumn !== columnName) {
+        clickCount = 0;
+        lastClickedColumn = columnName;
+      }
+
+      clickCount++;
+
+      if (clickCount === 1) {
+        // Перший клік
+        if (columnName === 'Name') {
+          sortASC(columnName, 0);
+        }
+
+        if (columnName === 'Position') {
+          sortASC(columnName, 1);
+        }
+
+        if (columnName === 'Office') {
+          sortASC(columnName, 2);
+        }
+
+        if (columnName === 'Age') {
+          sortASC(columnName, 3);
+        }
+
+        if (columnName === 'Salary') {
+          sortASC(columnName, 4);
+        }
+      } else if (clickCount === 2) {
+        if (columnName === 'Name') {
+          sortDESC(columnName, 0);
+        }
+
+        if (columnName === 'Position') {
+          sortDESC(columnName, 1);
+        }
+
+        if (columnName === 'Office') {
+          sortDESC(columnName, 2);
+        }
+
+        if (columnName === 'Age') {
+          sortDESC(columnName, 3);
+        }
+
+        if (columnName === 'Salary') {
+          sortDESC(columnName, 4);
+        }
```

Comment:
The sorting logic for each column is hardcoded. Consider refactoring this to make it more dynamic, especially if more columns are added in the future. This could involve mapping column names to their respective indices and sorting functions.